tools/online_editor/src/outline_widget.ts

```diff
@@ -0,0 +1,209 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+// cSpell: ignore lumino
+
+import { Message } from "@lumino/messaging";
+import { Widget } from "@lumino/widgets";
+
+import { MonacoLanguageClient } from "monaco-languageclient";
+
+import {
+  DocumentSymbolRequest,
+  DocumentSymbolParams,
+  DocumentSymbol,
+  SymbolInformation,
+} from "vscode-languageserver-protocol";
+
+const SYMBOL_KIND_MAP = [
+  "kind-unknown",
+  "kind-file",
+  "kind-module",
+  "kind-namespace",
+  "kind-package",
+  "kind-class",
+  "kind-method",
+  "kind-property",
+  "kind-field",
+  "kind-constructor",
+  "kind-enum",
+  "kind-interface",
+  "kind-function",
+  "kind-variable",
+  "kind-constant",
+  "kind-string",
+  "kind-number",
+  "kind-boolean",
+  "kind-array",
+  "kind-object",
+  "kind-key",
+  "kind-null",
+  "kind-enum-member",
+  "kind-struct",
+  "kind-event",
+  "kind-operator",
+  "kind-type-parameter",
+];
+
+function set_data(
+  data: DocumentSymbol[],
+  indent: number,
+  table: HTMLTableElement,
+  uri: string,
+  goto_position: (
+    _uri: string,
+    _start_line: number,
+    _start_character: number,
+    _end_line: number,
+    _end_column: number,
+  ) => void,
+) {
+  for (const d of data) {
+    const row = document.createElement("tr");
+    row.className = "outline-element";
+    if (d.deprecated || (d.tags != null && 1 in d.tags)) {
```

Comment:
is there not a constant you can use from some lsp library instead of `1`  ?
(and btw, deprecated is deprecated, and in slint we don't have deprecated symbol anyway so far)